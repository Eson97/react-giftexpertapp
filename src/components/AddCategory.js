import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = (({ setCategories }) => {

    const [inputValue, setInputValue] = useState('Nueva categorÃ­a');

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 0) {
            setCategories(cat => [inputValue, ...cat]);
            setInputValue('');
        }
    }

    const handleFocus = () => {
        setInputValue('');
    }

    const handleBlur = () =>{
        setInputValue('Nueva categoria');
    }

    return (
        <form onSubmit={handleSubmit} onFocus={handleFocus} onBlur={handleBlur}>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )

})

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
