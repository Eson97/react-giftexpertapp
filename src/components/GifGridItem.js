import React from 'react'



export const GifGridItem = ({title,url}) => {
    
    return (
        <div className="card animate__animated animate__fadeInLeft">
            <img src={url} alt={title}/>
            <p>{title}</p>
            <iframe title="share-button" src={`https://www.facebook.com/plugins/share_button.php?href=${url}&layout=button&size=large&width=103&height=28&appId`} width="103" height="28" style={{border:"none",overflow:"hidden",margin:"10px"}} scrolling="no" frameborder="0" allowfullscreen="true" allow="autoplay; clipboard-write; encrypted-media; picture-in-picture; web-share"></iframe>
        </div>
    )
}
