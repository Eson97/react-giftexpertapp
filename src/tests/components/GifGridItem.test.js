import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';


describe('Pruebas en <GifGridItem />>', () => {

    const title = 'Prueba titulo';
    const url = 'https://localhost/algo.jpg';

    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test('should show component correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should show title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })

    test('img should had url and alt from props', () => {
        const img = wrapper.find('img');
        // console.log(img.props()); //obtiene todos los props como objeto
        // console.log(img.prop('src')); //obtiene el prop especificado
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })

    test('div should had animate__fadeInLeft', () => {
        const div = wrapper.find('div');
        const res = div.prop('className').includes('animate__fadeInLeft');

        expect(res).toBe(true);


    })
    
    
})

