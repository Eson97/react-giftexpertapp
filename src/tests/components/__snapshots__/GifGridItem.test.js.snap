// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem />> should show component correctly 1`] = `
<div
  className="card animate__animated animate__fadeInLeft"
>
  <img
    alt="Prueba titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Prueba titulo
  </p>
  <iframe
    allow="autoplay; clipboard-write; encrypted-media; picture-in-picture; web-share"
    allowfullscreen="true"
    frameborder="0"
    height="28"
    scrolling="no"
    src="https://www.facebook.com/plugins/share_button.php?href=https://localhost/algo.jpg&layout=button&size=large&width=103&height=28&appId"
    style={
      Object {
        "border": "none",
        "margin": "10px",
        "overflow": "hidden",
      }
    }
    title="share-button"
    width="103"
  />
</div>
`;
